treeherder/services/intermittents_commenter/commenter.py

```diff
@@ -0,0 +1,284 @@
+    will be output to stdout rather than submitted to bugzilla"""
+
+    def __init__(self, weekly_mode, test_mode):
+        """booleans args are passed from the run_intermittents_commenter
+           command based on --weekly_mode and --test_mode flags; if flags
+           are omitted, default values for both are False"""
+
+        self.weekly_mode = weekly_mode
+        self.test_mode = test_mode
+
+    def run(self):
+        self.create_comments()
+
+    def create_comments(self):
+        """Posts a bug comment containing stats to each bug whose total number of
+           occurrences (daily or weekly) meet the appropriate threshold."""
+
+        startday, endday = self.calculate_date_strings(self.weekly_mode, 6)
+        bug_stats = self.get_bug_stats(startday, endday)
+        alt_startday, alt_endday = self.calculate_date_strings(True, DISABLE_DAYS)
+        alt_bug_stats = self.get_bug_stats(alt_startday, alt_endday)
+        test_run_count = self.get_test_runs(startday, endday)
+
+        template = Template(self.open_file('comment.template', False))
+        components = self.open_file('owner_triage_components.json', True)
+
+        session = self.new_request()
+
+        if self.weekly_mode:
+            top_bugs = [bug[0] for bug in sorted(bug_stats.items(), key=lambda x: x[1]['total'],
+                        reverse=True)][:TOP_BUGS_THRESHOLD]
+
+        for bug_id, counts in bug_stats.iteritems():
+            change_priority = None
+            bug_info = None
+            whiteboard = None
+            priority = 0
+            rank = None
+
+            if self.weekly_mode and bug_id in top_bugs:
+                rank = top_bugs.index(bug_id)+1
+
+            # recommend disabling when more than 150 failures tracked over 21 days
+            if alt_bug_stats[bug_id]['total'] >= DISABLE_THRESHOLD:
+                bug_info = self.fetch_bug_details(session, TRIAGE_PARAMS, bug_id)
+                if bug_info is not None:
+                    whiteboard = bug_info['whiteboard']
+                    if not self.check_whiteboard_status(whiteboard):
+                        priority = 3
+                        whiteboard = self.update_whiteboard(whiteboard, WHITEBOARD_DISABLE_RECOMMENDED)
+
+            if priority == 0 and self.weekly_mode:
+                if counts['total'] >= PRIORITY1_THRESHOLD:
+                    priority = 1
+                elif counts['total'] >= PRIORITY2_THRESHOLD:
+                    priority = 2
+
+            if priority == 2 or not self.weekly_mode:
+                if not bug_info:
+                    bug_info = self.fetch_bug_details(session, TRIAGE_PARAMS, bug_id)
+                    if bug_info is not None:
+                        whiteboard = bug_info['whiteboard']
+
+                        if (([bug_info['product'], bug_info['component']] in components) and
+                            not self.check_whiteboard_status(whiteboard)):
+
```

Comment:
I think a check for 'intermittent-failure' keyword was here originally. I think that's OK (either way, not a big deal).

Reply:
I left that out because the Bugscache model - used to store the bugzilla data for use in TH - only grabs bugs with the 'intermittent-failure' keyword.